### PR TITLE
Add homepage for developer blog on website

### PR DESCRIPTION
**Pull request by beechtom**
This adds a homepage that lists all developer blog posts in a single
location with short excerpts. It also lists posts in reverse
chronological order in the sidebar so new posts will be visible.

### CODE REVIEW

File: docs/_layouts/blog.html [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
```
Comment (nicklewis) [inline docs/_layouts/blog.html]: Why is this a layout and not just `docs/blog.html`?

File: docs/_layouts/default.html [hunk 2]
```diff
@@ -56,9 +56,10 @@
             </ul>
           </li>
           <li>
-            <span class="nav-header">Bolt Developer Updates</span>
+            <span class="nav-header">
+              <a href="{{ '/blog' | relative_url }}" class="center-vertical">Bolt Developer Updates</a></span>
             <ul>
-              {% for post in site.posts limit: 3 %}
+              {% for post in site.posts reversed limit: 3 %}
```
Comment (nicklewis) [inline docs/_layouts/default.html]: I don't think we want `reversed`. It seems like `posts` already sort from newest to oldest.

File: docs/_layouts/blog.html (hunk 1 above)
Comment (beechtom) [reply docs/_layouts/blog.html]: Updated.

File: docs/_layouts/default.html [hunk 3]
```diff
@@ -56,7 +56,8 @@
             </ul>
           </li>
           <li>
-            <span class="nav-header">Bolt Developer Updates</span>
+            <span class="nav-header">
+              <a href="{{ '/blog.html' | relative_url }}" class="center-vertical">Bolt Developer Updates</a></span>
```
Comment (nicklewis) [inline docs/_layouts/default.html]: The `.html` part doesn't seem to be necessary. I'm not sure I understand why though...

Comment (beechtom) [reply docs/_layouts/default.html]: Weird. Updated and also removed `.html` from the link to the debugging page while I was at it.

**Pull request merged**